libraries/AP_HAL_ChibiOS/hwdef/JHEMCU-GSF405A/hwdef.dat

```diff
@@ -0,0 +1,165 @@
+# hw definition file for processing by chibios_pins.py
+# JHEMCU GSF405A by V-22
+# With F405 MCU, MPU6000 IMU and MAX7456 series OSD
+# Based on Mamba F405 from jeanphilippehell
+@ thanks to betaflight for pin information
+
+MCU STM32F4xx STM32F405xx
+
+# board ID for firmware load
+APJ_BOARD_ID 1060
+
+# crystal frequency
+OSCILLATOR_HZ 8000000
+
+# board voltage
+STM32_VDD 330U
+
+# need to use timer 4 so timer 5 is available for bidir DShot
+define STM32_ST_USE_TIMER 4
+define CH_CFG_ST_RESOLUTION 16
+
+# order of I2C buses
+I2C_ORDER I2C1
+
+# order of UARTs (and USB)
+# this order follows the labels on the board
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 EMPTY USART6
+
+# The pins that USB is connected on
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# USB detection
+PA8 VBUS INPUT OPENDRAIN
+
+# USART1 (DMA) - wired to on-board ELRS receiver - need DMA for CRSF protocol
+PB6 USART1_TX USART1
+PA10 USART1_RX USART1
+
+# default Serial1 to ELRS RX
+define HAL_SERIAL1_PROTOCOL 23
+define HAL_RSSI_TYPE 3 
+
+# Alt config to allow PPM on RX1
+#PA10 TIM1_CH3 TIM1 RCININT PULLDOWN ALT(1)
+
+# USART2
+PD5 USART2_TX USART2 NODMA
```

Comment:
This really needs to have DMA since the docs say this is where to connect an external CRSF receiver

Reply:
Is it OK for SPI3 and I2C1 to share DMA resources? If so, we can move the motor 3 and 4 outputs from TIM5 to TIM2. We'd lose DMA on USART3.

`
# USART2
PD5 USART2_TX USART2
PD6 USART2_RX USART2

# USART3
PB10 USART3_TX USART3 NODMA
PB11 USART3_RX USART3 NODMA

# Motor outputs on AIO ESC
PB0 TIM3_CH3 TIM3 PWM(1) GPIO(50) 
PB1 TIM3_CH4 TIM3 PWM(2) GPIO(51) BIDIR
PA3 TIM2_CH4 TIM2 PWM(3) GPIO(52) 
PA2 TIM2_CH3 TIM2 PWM(4) GPIO(53) BIDIR

DMA_PRIORITY SPI1* SPI3* TIM1_CH2 TIM2_UP TIM3_UP
DMA_NOSHARE USART2*`

I think the default configuration should be as proposed since the ELRS receiver is built-in, which would require a second target for DMA on USART2.

Reply:
Maybe - I guess as long as you are thorough with the docs its ok.

SPI3 sharing with I2C is probably ok - the hog here is flash which has a very high duty cycle, but I2C is pretty slow so probably not much of a problem. You would have to test to verify.

Reply:
See my comments above about SBUS

Reply:
I created a second target for the case where DMA is needed on USART2 for an external receiver.